nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,467 @@
+# 3. success: (sparkles / party_popper).
+clipboard_emoji = "📋"
+sad_cat_emoji = "😿"
+broom_emoji = "🧹"
+doughnut_emoji = "🍩"
+sparkles_emoji = "✨"
+rocket_emoji = "🚀"
+bagel_emoji = "🥯"
+party_popper_emoji = "🎉"
+confetti_emoji = "🎊"
+
+# based on quantity
+medium_throughput = 100
+sushi_emoji = "🍣"  # 100 subjects in BIDS
+shaved_ice_emoji = "🍧"  # 100 subjects processed
+
+# TODO secret rewards based on high-throuput and update time
+surprise_box_emoji = "🎁"
+high_throughput = 1000
+bento_box_emoji = "🍱"  # 1000 subjects in BIDS
+mate_emoji = "🧉"  # 1000 subjects processed
+night_owl_emoji = "🦉"
+slow_sloth_emoji = "🦥"
+quick_tiger_emoji = "🐅"
+
+secret_rewards_dict = {
+    bento_box_emoji: "1000+ subjects in BIDS! That's impressive attention to detail!",
+    mate_emoji: "1000+ subjects are processed. Deserves a drink!",
+    night_owl_emoji: "Night owl! That's some dedication!",
+    slow_sloth_emoji: "Slow sloth! This dataset could use some attention!",
+    quick_tiger_emoji: "Quick tiger! You are blazing through!",
+    surprise_box_emoji: "There are more secret rewards to be unlocked!",
+}
+
+# emoji legend
+legend_dict = {
+    clipboard_emoji: "Manifest entries available!",
+    sad_cat_emoji: "No available data are curated or processed",
+    broom_emoji: "Data clean-up (i.e. BIDSification) has started!",
+    doughnut_emoji: "Curation in progress... Doughnut is made!",
+    sparkles_emoji: "BIDSification is complete! Tidiness sparks joy!",
+    rocket_emoji: "Data processing has launched successfully!",
+    bagel_emoji: "Data processing in progress... Bagels are baking!",
+    party_popper_emoji: "At least 1 pipeline processing is complete!",
+    confetti_emoji: "Amazing! All data are curated and processed. Time to celebrate!",
+    sushi_emoji: "100+ subjects in BIDS! Meticulousness should be rewarded!",
+    shaved_ice_emoji: "100+ subjects are processed. Deseves a cool treat!",
+}
+
+
+class StatusWorkflow(BaseWorkflow):
+    """Workflow for status command."""
+
+    def __init__(
+        self,
+        dpath_root: Path,
+        fpath_layout: Optional[StrOrPathLike] = None,
+        logger: Optional[logging.Logger] = None,
+        dry_run: bool = False,
+    ):
+        """Initialize the workflow."""
+        super().__init__(
+            dpath_root=dpath_root,
+            name="status",
+            fpath_layout=fpath_layout,
+            logger=logger,
+            dry_run=dry_run,
+        )
+        self.col_pipeline = "pipeline"
+
+    def run_main(self):
+        """Check the status of the dataset and report.
+
+        1) Number of participants in manifest per BIDS datatype,
+        2) Doughnut information if available,
+        3) Bagel information if available
+        """
+        self.logger.info("Checking the status of the dataset.")
+
+        # load global_config to get the dataset name
+        dataset_name = self.config.DATASET_NAME
+        expected_sessions = self.config.SESSION_IDS
```

Comment:
Or: we should have the visit/session ids printed out during the manifest check follow the same order as what is in the manifest. I kind of like that better because it lets users define their own preferred order. But this would require #427 to be addressed first